mantidimaging/gui/windows/live_viewer/view.py

```diff
@@ -55,8 +56,14 @@ def remove_image(self) -> None:
         """Remove the image from the view."""
         self.live_viewer.handle_deleted()
 
+    def set_image_range(self, index_range: tuple[int, int]) -> None:
+        with QSignalBlocker(self.live_viewer.z_slider):
+            self.live_viewer.z_slider.set_range(*index_range)
+
     def set_image_index(self, index: int) -> None:
-        self.live_viewer.z_slider.set_value(index)
+        with QSignalBlocker(self.live_viewer.z_slider):
+            self.live_viewer.z_slider.set_value(index)
+        self.live_viewer.z_slider.valueChanged.emit(index)
```

Comment:
NITPICK: I would probably want docstring on both of these to explain why we need to use `QSignalBlocker` so there's no possibility of removing later down the line when the context for them being added is forgotten.